### PR TITLE
Wrap the realtime RC test in flaky-block to automatically retry.

### PR DESCRIPTION
**Pull request by AlmostMatt**
### Description
Flaky block will automatically retry the whole test.
Since the test only occasionally times out, this retry is likely to succeed.
### Testing
Ran integration test on github runner several times
https://github.com/firebase/firebase-cpp-sdk/actions/runs/5683195628 
https://github.com/firebase/firebase-cpp-sdk/actions/runs/5684526029 
https://github.com/firebase/firebase-cpp-sdk/actions/runs/5684529810 
https://github.com/firebase/firebase-cpp-sdk/actions/runs/5684532551 
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: remote_config/integration_test/src/integration_test.cc [hunk 1]
```diff
@@ -266,6 +266,7 @@ TEST_F(FirebaseRemoteConfigTest, TestSetDefault) {
 TEST_F(FirebaseRemoteConfigTest, TestAddOnConfigUpdateListener) {
   ASSERT_NE(rc_, nullptr);
 
+  FLAKY_TEST_SECTION_BEGIN();
```
Comment (jonsimantov) [inline remote_config/integration_test/src/integration_test.cc:269]: As a reminder, does this flake on all platforms? If not you can surround these macros with #ifdefs ...

File: remote_config/integration_test/src/integration_test.cc [hunk 2]
```diff
@@ -302,6 +303,7 @@ TEST_F(FirebaseRemoteConfigTest, TestAddOnConfigUpdateListener) {
          firebase::remote_config::RemoteConfigError) {});
 #else
   auto config_update_promise = std::make_shared<std::promise<void> >();
+  auto config_update_future = config_update_promise->get_future();
```
Comment (jonsimantov) [inline remote_config/integration_test/src/integration_test.cc:306]: Is all this promise stuff just used for the delay?

File: remote_config/integration_test/src/integration_test.cc [hunk 3]
```diff
@@ -333,6 +334,7 @@ TEST_F(FirebaseRemoteConfigTest, TestAddOnConfigUpdateListener) {
     registration.Remove();
```
Comment (jonsimantov) [inline remote_config/integration_test/src/integration_test.cc:334]: nit (a few lines above): key_valur_pair --> key_value_pair

**Review by jonsimantov — APPROVED** [review]
Looks good, just a few nits. Also, if there is a smaller range of the test that can be surrounded by the flaky test macros, could you narrow the flaky section? (If not, it's fine.)

File: remote_config/integration_test/src/integration_test.cc (hunk 2 above)
Comment (AlmostMatt) [reply remote_config/integration_test/src/integration_test.cc:306]: Yea the promise+future is used just to verify that the callback is called.
The promise is used to set a value, and the corresponding future is used to read the value once it has been set.

File: remote_config/integration_test/src/integration_test.cc (hunk 1 above)
Comment (AlmostMatt) [reply remote_config/integration_test/src/integration_test.cc:269]: The initial bug report is for the test being flaky on Android. I'll add an IF ANDROID check here and if it ends up flaking on other platforms we can change it.

File: remote_config/integration_test/src/integration_test.cc (hunk 3 above)
Comment (AlmostMatt) [reply remote_config/integration_test/src/integration_test.cc:334]: Done, did a find replace on the typo as it was copied from another test case.

**Review by jonsimantov — APPROVED** [review]

**Pull request merged**